pkg/entrypoint/entrypointer.go

```diff
@@ -142,19 +145,46 @@ func (e Entrypointer) Go() error {
 	}
 
 	if err == nil {
-		ctx := context.Background()
 		var cancel context.CancelFunc
 		if e.Timeout != nil && *e.Timeout != time.Duration(0) {
 			ctx, cancel = context.WithTimeout(ctx, *e.Timeout)
-			defer cancel()
+		} else {
+			ctx, cancel = context.WithCancel(ctx)
+		}
+		defer cancel()
+
+		errChan := make(chan error, 1)
+		go func() {
+			errChan <- e.Runner.Run(ctx, e.Command...)
+			cancel()
```

Comment:
Does it mean we'll run `cancel` twice ? (here and the `defer`)
I don't remember if it's a no-op or if it panics.. If it's a no-op we are fine though.

Reply:
Thank you for the review. Yes, we will end up calling `cancel` twice on the context, but that is no-op ([I tested to be sure](https://go.dev/play/p/PKt2M6X7inR)).
I added the `defer cancel()` for both readability and ensuring clean-up. If removing it is preferred, please let me know.